### PR TITLE
Add signal handler for process execution

### PR DESCRIPTION
**Pull request by ivanduplenskikh**
**Changes**: Added signal handler for process execution to kill process with proper signal. This PR also adds unit tests.

**Related**: [AB#2137168](https://dev.azure.com/mseng/b924d696-3eae-4116-8443-9a18392d8544/_workitems/edit/2137168)

### CODE REVIEW

File: node/toolrunner.ts [hunk 1]
```diff
@@ -1312,18 +1314,20 @@ export class ToolRunner extends events.EventEmitter {
...
         });
 
-        cp.on('close', (code: number, signal: any) => {
+        cp.on('close', (code: number, signal: number | NodeJS.Signals) => {
             state.processExitCode = code;
             state.processExited = true;
             state.processClosed = true;
-            this._debug(`STDIO streams have closed for tool '${this.toolPath}'`)
+            this._debug(`STDIO streams have closed for tool '${this.toolPath}'`);
+            this._debug(`Signal ${signal} received from tool '${this.toolPath}'`);
```
Comment (KonstantinTyukalov) [inline node/toolrunner.ts:1330]: Suggestion:
```suggestion
            this._debug(`STDIO streams have closed and received signal ${signal} for tool '${this.toolPath}'`);
```

And the same for others. What do you think?

**Review by KonstantinTyukalov — APPROVED** [review]

File: node/toolrunner.ts (hunk 1 above)
Comment (ivanduplenskikh) [reply node/toolrunner.ts:1330]: Sure, I'll fix output

Comment (ivanduplenskikh) [reply node/toolrunner.ts]: I've also added exit code to output

**Review by DmitriiBobreshev — APPROVED** [review]

**Review by kirill-ivlev — APPROVED** [review]

**Pull request merged**